src/Clapi/Relay.hs

```diff
@@ -143,14 +144,29 @@ treeDeltaToMsg p td = case td of
 deltaToMsg :: VsDelta -> Message
 deltaToMsg (p, d) = either (MsgAssignType p) (treeDeltaToMsg p) d
 
+modifyRootTypePath :: Valuespace Validated -> Valuespace Unvalidated -> Valuespace Unvalidated
+modifyRootTypePath vs vs' = if rootType' == rootType then vs' else vs''
+  where
+    rootTypePath = moe "root has no type" $ getType vs' []
+    rootTypeNode = loe "root type path missing" rootTypePath $ vsGetTree vs'
+    (a, mtp) = loe "root type missing zero" z $ loe "root type missing global site" Nothing $ _getSites rootTypeNode
+    (i, rootType) = moe "root type deleted" mtp
+    rootType' = foldl updateRootType rootType $ mapMaybe nsChange $ vsDiff vs vs'
+    vs'' = moe "root type set failed" $ vsSet a i rootType' rootTypePath Nothing z vs'
+    z = Time 0 0
+    loe s k m = moe s $ Map.lookup k m
+    moe s m = maybe (error s) id m
```

Comment:
Is this a hard `error` because our code should never hit this path in production?

Reply:
I think so, the only way we should be able to end up here is by the baseValuespace being wrong or this function being buggy.